aws/table_aws_cloudtrail_query.go

```diff
@@ -0,0 +1,242 @@
+				},
+				{
+					Name:    "query_status",
+					Require: plugin.Optional,
+				},
+				{
+					Name:      "creation_time",
+					Require:   plugin.Optional,
+					Operators: []string{"=", "<=", "<", ">", ">="},
+				},
+			},
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "query_id",
+				Description: "The ID of the query.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "event_data_store_arn",
+				Description: "The ID of the event data store.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "creation_time",
+				Description: "The creation time of the query.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("CreationTime", "QueryStatistics.CreationTime"),
+			},
+			{
+				Name:        "query_status",
+				Description: "The status of a query. Values for QueryStatus include QUEUED, RUNNING, FINISHED, FAILED, TIMED_OUT, or CANCELLED.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "bytes_scanned",
+				Description: "Gets metadata about a query, including the number of events that were matched, the total number of events scanned, the query run time in milliseconds, and the query's creation time.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudTrailLakeQuery,
+				Transform:   transform.FromField("QueryStatistics.BytesScanned"),
+			},
+			{
+				Name:        "events_matched",
+				Description: "The number of events that matched a query.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudTrailLakeQuery,
+				Transform:   transform.FromField("QueryStatistics.EventsMatched"),
+			},
+			{
+				Name:        "events_scanned",
+				Description: "The number of events that the query scanned in the event data store.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudTrailLakeQuery,
+				Transform:   transform.FromField("QueryStatistics.EventsScanned"),
+			},
+			{
+				Name:        "execution_time_in_millis",
+				Description: "The query's run time, in milliseconds.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     getCloudTrailLakeQuery,
+				Transform:   transform.FromField("QueryStatistics.ExecutionTimeInMillis"),
+			},
+			{
+				Name:        "query_string",
+				Description: "The SQL code of a query.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getCloudTrailLakeQuery,
+			},
+
+			// steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("QueryId"),
+			},
+		}),
+	}
+}
+
+// We need custom structure
+// 1. EventDataStoreArn is required for joing the queries
+// 2. list/get call do not provide the EventDataStoreArn
+// 3. Both list/get API call return different stracture with dfferent data
+type QueryInfo struct {
```

Comment:
@ParthaI Instead of having just 1 custom structure, could we have 2 structures instead, 1 for List results and 1 for Get results? Each of these would include the event data store ARN and then also contain the AWS SDK structure related to the List or Get call (so we don't have to redefine each property).

I think this would allow us to lean into the structures AWS has defined while adding any properties, like the event data store as well.

Another advantage would be it would be less likely to miss columns, like we are today, as the DescribeQuery response has attributes we don't have as columns today. So any missing columns, like the delivery status, delivery S3 URI, and error message should be added.

Our columns would then use Transform.FromFields and leverage the ability to specify multiple paths for the columns that are returned from the List and Get calls.

Reply:
We were using `v1.16.8` of Cloudtrail package, DescribeQuery API was not returning Delivery status, Delivery S3 URI attributes with this package. Now I have upgraded package to `v1.20.4` and added those missing columns.